tests/ui_tests/app/page/admin/page.py

```diff
@@ -570,35 +570,49 @@ def fill_select_in_policy_popup(self, items, available_items):
+                self.fill_select_in_policy_popup(clusters, self.find_elements(AdminPoliciesLocators.item))
+                self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.SecondStep.cluster_select)
         if services:
             if not parent:
                 raise ValueError("There are should be parent for service")
             with allure.step(f"Select services {services} in popup"):
+                self.wait_element_visible(AdminPoliciesLocators.AddPolicyPopup.SecondStep.service_select)
                 self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.SecondStep.service_select)
                 self.wait_element_visible(AdminPoliciesLocators.AddPolicyPopup.SecondStep.service_item)
-                available_services = self.find_elements(AdminPoliciesLocators.AddPolicyPopup.SecondStep.service_item)
-                self.fill_select_in_policy_popup(services, available_services)
-                fill_cluster_select(parent)
+                self.fill_select_in_policy_popup(
+                    services, self.find_elements(AdminPoliciesLocators.AddPolicyPopup.SecondStep.service_item)
+                )
+
+                self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.SecondStep.parent_select)
+                self.wait_element_visible(AdminPoliciesLocators.item)
+                self.fill_select_in_policy_popup(parent, self.find_elements(AdminPoliciesLocators.item))
+                self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.SecondStep.parent_select)
+        if hosts:
+            with allure.step(f"Select hosts {hosts} in popup"):
+                self.wait_element_visible(AdminPoliciesLocators.AddPolicyPopup.SecondStep.hosts_select)
+                self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.SecondStep.hosts_select)
+                self.wait_element_visible(AdminPoliciesLocators.item)
+                self.fill_select_in_policy_popup(hosts, self.find_elements(AdminPoliciesLocators.item))
+                self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.SecondStep.hosts_select)
         if providers:
-            self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.SecondStep.provider_select)
-            self.wait_element_visible(AdminPoliciesLocators.item)
-            available_providers = self.find_elements(AdminPoliciesLocators.item)
-            self.fill_select_in_policy_popup(providers, available_providers)
-            self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.SecondStep.provider_select)
+            with allure.step(f"Select providers {providers} in popup"):
+                self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.SecondStep.provider_select)
+                self.wait_element_visible(AdminPoliciesLocators.item)
+                self.fill_select_in_policy_popup(providers, self.find_elements(AdminPoliciesLocators.item))
+                self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.SecondStep.provider_select)
```

Comment:
I suggest moving such blocks to a separate functions as it was with clusters

Reply:
yes, changed, thanks!